// pages/EditArticle/EditArticle.js
import React, { useEffect, useState } from 'react'
import ArticleForm from 'components/ArticleForms/ArticleForms'
import useFetch from 'hooks/useFetch'
import { Redirect } from 'react-router-dom'
const EditArticle = ({match}) => {
    const slug = match.params.slug
    const apiUrl = `/articles/${slug}`
    const [{ response: fetchArticleResponse}, doFetchArticle ]=useFetch(apiUrl)
    const [{response:updateArticleResponse , error: updateArticleError}, doUpadateArticle]= useFetch(apiUrl)
    const [initialValues,setInitialValues] = useState(null)
    const [isSuccessfullSumbit, setIsSuccessfullSubmit]= useState(false)
    const handelSubmit = article => {
        doUpadateArticle({
            method: 'put',
            data: {
                article
            }
        })
    }


    useEffect(()=>{
        doFetchArticle()
    },[doFetchArticle])
    
    useEffect(()=> {
        if (!fetchArticleResponse){
        return
    }
        setInitialValues({
            title: fetchArticleResponse.article.title,
            description: fetchArticleResponse.article.description,
            body: fetchArticleResponse.article.body,
            tagList: fetchArticleResponse.article.tagList
        })
       
    },[fetchArticleResponse])
    useEffect(()=> {
        if(!updateArticleResponse){
            return
        }
        setIsSuccessfullSubmit(true)
    },[updateArticleResponse])
    if (isSuccessfullSumbit){
        return <Redirect to={`/articles/${slug}`}/>
    }


    return(
        <ArticleForm
        onSubmit={handelSubmit}
        errors={(updateArticleError && updateArticleError.errors)|| {}}
        initialValues = {initialValues}
        />
    )
}
export default EditArticle
