// pages/creatArticle/creatArticle.js
import React, { useEffect, useState } from 'react'
import ArticleForm from 'components/ArticleForms/ArticleForms'
import useFetch from 'hooks/useFetch'
import { Redirect } from 'react-router-dom'

const CreatArticle = () => {
    const apiUrl = '/articles'
    const [{response,error}, doFetch] = useFetch(apiUrl)
    const [isSuccessfullSubmit, setIsSuccssfullSubmit] = useState(false)
    const errors={}
    const initialValues ={
        title: '',
        description: '',
        body: '',
        tagList: []
    }
    const handleSubmit = article => {
        doFetch({
            method: 'post',
            data: {
               article 
            }
        })
    }
    useEffect(()=>{
        if(!response){
            return
        }
        setIsSuccssfullSubmit(true)
    },[response])
    if(isSuccessfullSubmit){
        return <Redirect to={`/articles/${response.article.slug}`}/>
    }
    return(
        <div><ArticleForm
        errors={(error && error.errors)|| {}}
        initialValues={initialValues}
        onSubmit={handleSubmit}
        /></div>
    )
}

export default CreatArticle
